transaction/atomic.go

```diff
@@ -119,23 +116,47 @@ func AtomicTransfer(hezClient client.HermezClient, ethereumChainID int,
 				position = currentAtomicTxId - 1
 			}
 		}
-		atomicGroup.Txs[currentAtomicTxId].RqFromIdx = atomicGroup.Txs[position].FromIdx
-		atomicGroup.Txs[currentAtomicTxId].RqToIdx = atomicGroup.Txs[position].ToIdx
-		atomicGroup.Txs[currentAtomicTxId].RqToEthAddr = atomicGroup.Txs[position].ToEthAddr
-		atomicGroup.Txs[currentAtomicTxId].RqToBJJ = atomicGroup.Txs[position].ToBJJ
-		atomicGroup.Txs[currentAtomicTxId].RqNonce = atomicGroup.Txs[position].Nonce
-		atomicGroup.Txs[currentAtomicTxId].RqFee = atomicGroup.Txs[position].Fee
-		atomicGroup.Txs[currentAtomicTxId].RqAmount = atomicGroup.Txs[position].Amount
-		atomicGroup.Txs[currentAtomicTxId].RqTokenID = atomicGroup.Txs[position].TokenID
-		atomicGroup.Txs[currentAtomicTxId].RqOffset = uint8(txs[currentAtomicTxId].RqOffSet)
+		fullTxs[currentAtomicTxId].RqFromIdx = fullTxs[position].FromIdx
+		fullTxs[currentAtomicTxId].RqToIdx = fullTxs[position].ToIdx
+		fullTxs[currentAtomicTxId].RqToEthAddr = fullTxs[position].ToEthAddr
+		fullTxs[currentAtomicTxId].RqToBJJ = fullTxs[position].ToBJJ
+		fullTxs[currentAtomicTxId].RqNonce = fullTxs[position].Nonce
+		fullTxs[currentAtomicTxId].RqFee = fullTxs[position].Fee
+		fullTxs[currentAtomicTxId].RqAmount = fullTxs[position].Amount
+		fullTxs[currentAtomicTxId].RqTokenID = fullTxs[position].TokenID
+		fullTxs[currentAtomicTxId].RqOffset = uint8(txs[currentAtomicTxId].RqOffSet)
 	}
+	return
+}
 
+// SetAtomicGroupId defines the AtomicGroup ID and propagate to txs
+func SetAtomicGroupId(atomicGroup api.AtomicGroup) api.AtomicGroup {
 	// Generate atomic group id
 	atomicGroup.SetAtomicGroupID()
 
+	for currentAtomicTxId := range atomicGroup.Txs {
+		atomicGroup.Txs[currentAtomicTxId].AtomicGroupID = atomicGroup.ID
+	}
+	return atomicGroup
+}
+
+// AtomicTransfer perform token or ETH transfers in a pool of transactions
```

Comment:
Function documentation should explain that txs are going to be signed within this method and sent to the current coordinator